front/app/containers/App/index.tsx

```diff
@@ -213,11 +215,26 @@ const App = ({ children }: Props) => {
 
   useEffect(() => {
     const subscriptions = [
-      locale$.subscribe((locale) => {
-        const momentLoc = appLocalesMomentPairs[locale] || 'en';
-        moment.locale(momentLoc);
-        setLocale(locale);
+      locale$.subscribe(async (locale) => {
+        try {
+          const momentLoc = appLocalesMomentPairs[locale] || 'en';
+
+          // No need to import for English
+          if (momentLoc !== 'en') {
+            await localeGetter(momentLoc);
+          }
```

Comment:
Is this necessary since we've already done this importing higher up? This whole locale subscription block looks like it's not doing anything useful at the moment. (The locale in the state can be replaced with a useLocale hook.)

Reply:
> This whole locale subscription block looks like it's not doing anything useful at the moment.

With this I meant that I commented it out, added the useLocale hook to replace locale state, and everything worked the same. So the moment code doesn't seem to be effective here.

Reply:
Yeah, got it. But doing the moment updates in the subscription doesn't seem to work well everywhere. So I'd like to work on this bit.